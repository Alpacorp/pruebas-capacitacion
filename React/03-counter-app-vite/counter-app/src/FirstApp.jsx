import PropTypes from "prop-types";

export const FirstApp = ({ title, subtitle, name }) => {
  return (
    <>
      <h1>{title}</h1>
      <p>{subtitle}</p>
      <p>{name}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
};

FirstApp.defaultProps = {
  name: "Alejandro Palacios",
  subtitle: "No hay subtítulo",
  title: "No hay título",
};
