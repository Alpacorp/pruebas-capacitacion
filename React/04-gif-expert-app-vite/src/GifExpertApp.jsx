import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

const GifExpertApp = () => {
  const [categories, setCategories] = useState(["God of War"]);

  const onAddCategory = (newCategory) => {
    const lowerCaseCategory = newCategory.toLowerCase();
    const existentCategory = categories.map((category) =>
      category.toLowerCase()
    );

    if (existentCategory.includes(lowerCaseCategory)) return;
    setCategories((category) => [newCategory, ...category]);
  };

  return (
    <div>
      <h1>GifExpertApp</h1>
      <AddCategory
        onNewCategory={(newCategory) => onAddCategory(newCategory)}
      />
      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </div>
  );
};

export default GifExpertApp;
