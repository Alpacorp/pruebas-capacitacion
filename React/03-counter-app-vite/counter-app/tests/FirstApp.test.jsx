/** @jest-environment jsdom */

import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Pruebas en <FirsApp />", () => {
  test("debe hacer match con el snapshot", () => {
    const title = "Hola, soy Alejo";
    const { container } = render(<FirstApp title={title} />);

    expect(container).toMatchSnapshot();
  });
});
