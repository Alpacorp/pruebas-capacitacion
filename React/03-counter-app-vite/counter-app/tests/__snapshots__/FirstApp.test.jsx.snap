// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirsApp /> debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, soy Alejo
  </h1>
  <p>
    No hay subt√≠tulo
  </p>
  <p>
    Alejandro Palacios
  </p>
</div>
`;
